packages/safe-tools-client/tests/integration/future-payments-list-test.ts

```diff
@@ -52,36 +101,50 @@ class ScheduledPaymentsStub extends Service {
         tokenAddress: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
         chainId,
         payeeAddress: '0xeBCC5516d44FFf5E9aBa2AcaeB65BbB49bC3EBe1',
-        payAt: addMonths(minPayAt, 1),
+        payAt: endOfThisMonth,
       },
       {
-        amount: '20000000',
+        amount: '11000000',
         feeFixedUSD: '0',
         feePercentage: '0',
         gasTokenAddress: '0x123',
         tokenAddress: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
         chainId,
         payeeAddress: '0xeBCC5516d44FFf5E9aBa2AcaeB65BbB49bC3EBe1',
-        payAt: addMonths(minPayAt, 3),
+        payAt: addDays(endOfThisMonth, 1),
       },
-    ]);
+    ];
+
+    if (returnScheduledPaymentsUntilTomorrow) {
+      return Promise.resolve(
+        scheduledPayments.filter((sp) => sp.payAt <= endOfTomorrow)
+      );
+    }
+
+    return Promise.resolve(scheduledPayments);
   };
 }
 
 module('Integration | Component | future-payments-list', function (hooks) {
   setupRenderingTest(hooks);
+  setupFakeDateService(hooks);
 
-  hooks.beforeEach(function (this: TestContext) {
+  hooks.beforeEach(async function (this: TestContext) {
     this.owner.register('service:wallet', WalletServiceStub);
     this.owner.register('service:scheduled-payments', ScheduledPaymentsStub);
     this.owner.register(
       'service:hub-authentication',
       HubAuthenticationServiceStub
     );
+
+    dateService = this.owner.lookup('service:date') as FakeDateService;
+    await dateService.setNow(NOW.getTime());
```

Comment:
Is redefining "now" really needed? If you use the real current timestamps the test should work anyway right?

Reply:
Oh I see. Makes sense